### PR TITLE
Add a release to PyPI workflow

### PR DESCRIPTION
**Pull request by ekohl**
This uploads releases to PyPI which is easier in packaging workflows. It also adds a test in CI to actually build the package to make sure that part works.

This adds an URL which and removes the empty requires variable in favor an inline definition.

### CODE REVIEW

**Comment by jturel** [comment]
Needs a rebase and I guess it's ready for merge

**Comment by ekohl** [comment]
Rebased

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -2,6 +2,21 @@ name: build
 on: [pull_request]
 
 jobs:
+  build:
```
Comment (jturel) [inline .github/workflows/pull_request.yml:5]: Looking more closely now - does this job do anything meaningful?

Comment (jturel) [reply .github/workflows/pull_request.yml:5]: Nvm - I see the build output now :)

**Review by jturel — APPROVED** [review]
ACK

**Pull request merged**